python/pyspark/ml/classification.py

```diff
@@ -60,6 +61,137 @@ def numClasses(self):
 
 
 @inherit_doc
+class LinearSVC(JavaEstimator, HasFeaturesCol, HasLabelCol, HasPredictionCol, HasMaxIter,
+                HasRegParam, HasTol, HasRawPredictionCol, HasFitIntercept, HasStandardization,
+                HasThreshold, HasWeightCol, HasAggregationDepth, JavaMLWritable, JavaMLReadable):
+    """
+    Linear SVM Classifier (https://en.wikipedia.org/wiki/Support_vector_machine#Linear_SVM)
+    This binary classifier optimizes the Hinge Loss using the OWLQN optimizer.
+
+    >>> from pyspark.sql import Row
+    >>> from pyspark.ml.linalg import Vectors
+    >>> bdf = sc.parallelize([
+    ...     Row(label=1.0, weight=2.0, features=Vectors.dense(1.0)),
+    ...     Row(label=0.0, weight=2.0, features=Vectors.sparse(1, [], []))]).toDF()
+    >>> svm = LinearSVC(maxIter=5, regParam=0.01, weightCol="weight")
+    >>> model = svm.fit(bdf)
+    >>> model.coefficients
+    DenseVector([1.909])
+    >>> model.intercept
+    -1.0045358384178
+    >>> model.numClasses
+    2
+    >>> model.numFeatures
+    1
+    >>> test0 = sc.parallelize([Row(features=Vectors.dense(-1.0))]).toDF()
+    >>> result = model.transform(test0).head()
+    >>> result.prediction
+    0.0
+    >>> result.rawPrediction
+    DenseVector([2.9135, -2.9135])
+    >>> test1 = sc.parallelize([Row(features=Vectors.sparse(1, [0], [1.0]))]).toDF()
+    >>> model.transform(test1).head().prediction
+    1.0
+    >>> svm.setParams("vector")
```

Comment:
I know, there are some not great examples to follow.  It'd be nice to clean those out sometime...